### PR TITLE
[SIG-4375] Nearby incidents

### PR DESCRIPTION
**Pull request by janjaap**
Adds the `Nearby` component. This component can only be rendered in the `AssetSelectProvider` context and will plot markers on a map by current bounding box and category and subcategory settings from the incident state.
Note that, due to limitations with the DOM in the test environment, markers cannot be rendered on a canvas plane and, as such, their visibility cannot be asserted in the tests in this PR.

### CODE REVIEW

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.test.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    ).not.toHaveBeenCalled()
+
+    renderWithContext()
+
+    await screen.findByTestId('nearby')
+
+    expect(
+      Leaflet.FeatureGroup.prototype.clearAllEventListeners
+    ).toHaveBeenCalled()
+    // expect(mockFeatureGroup.clearLayers).toHaveBeenCalled()
```
Comment (kiruys) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.test.tsx:146]: why is this here?

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.test.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+  it('clears the feature group when the API returns an error ', async () => {
+    jest.mocked(useFetch).mockImplementation(() => ({
+      ...useFetchResponse,
+      data: undefined,
+      error: true,
+    }))
+
+    expect(mockUseMapInstance.removeLayer).not.toHaveBeenCalled()
+    // expect(mockFeatureGroup.clearLayers).not.toHaveBeenCalled()
```
Comment (kiruys) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.test.tsx:134]: why is this here? can it be removed?

Comment (janjaap) [reply src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.test.tsx:134]: Forgot to remove. Will remove it.

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.test.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    Leaflet.Marker.prototype.setIcon = jest.fn()
+
+    jest
+      .spyOn(reactRedux, 'useSelector')
+      .mockReturnValue({ category, subcategory })
+
+    jest.mocked(useFetch).mockImplementation(() => useFetchResponse)
+  })
+
+  it('renders a span placeholder for test await and QA purposes', () => {
```
Comment (kiruys) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.test.tsx:87]: what are QA purposes?

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.tsx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  const { get, data, error } = useFetch<FeatureCollection<Point, Properties>>()
+  const [activeLayer, setActiveLayer] = useState<NearbyMarker>()
+  const featureGroup = useRef<L.FeatureGroup<NearbyMarker>>(L.featureGroup())
+
+  const onMarkerClick = useCallback(
+    /* istanbul ignore next */ (feature: Feature<Point, Properties>) =>
+      async ({ sourceTarget }: MarkerMouseEvent) => {
+        sourceTarget.setIcon(nearbyMarkerSelectedIcon)
+
+        setActiveLayer(sourceTarget)
```
Comment (kiruys) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.tsx:91]: To me the term `activeLayer` implies that it can contain more than one marker. Is that also your conception? Or do you mean the marker that is selected?

File: src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.tsx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    if (element !== targetElement) {
+      layer.setIcon(nearbyMarkerIcon)
+    }
+  })
+}
+
+/**
+ * Component using the Leaflet API to plot markers in a feature layer on a map
+ * Because the markers are rendered in a feature group, events can be captured by that group.
+ * Which is useful, because the all of the layers in the feature group can be easily targeted.
```
Comment (kiruys) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.tsx:75]: What is the purpose of the featureGroupClick? Can you specify an example where all of the layers need to be targeted at once?
